File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPWalkSpeed.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPWalkSpeed implements PlayerStateProperty<Float> {
    @Override
    public String getConfigName() {
        return "walk_speed";
    }

    @Override
    public Float getDefaultSerializedProperty() {
        return 0.2F;
    }

    @Override
    public Float getSerializedPropertyFromPlayer(Player player) {
        return player.getWalkSpeed();
    }

    @Override
    public void restoreProperty(Player player, Float speed) {
        player.setWalkSpeed(speed);
    }

    @Override
    public Class<Float> getTypeClass() {
        return Float.class;
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPWalkSpeed.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPWalkSpeed extends PlayerStateProperty[Float]:
  override def getConfigName: String = "walk_speed"

  override def getDefaultSerializedProperty: Float = 0.2

  override def getSerializedPropertyFromPlayer(player: Player): Float =
    player.getWalkSpeed

  override def restoreProperty(player: Player, speed: Float): Unit =
    player.setWalkSpeed(speed)

  override def getValueFromConfig(config: ConfigurationSection): Float =
    config.getDouble(getConfigName).toFloat